bundles/sirix-core/src/main/java/org/sirix/access/DatabaseManager.java

```diff
@@ -0,0 +1,22 @@
+package org.sirix.access;
+
+import dagger.Component;
+import org.sirix.api.json.JsonResourceManager;
+import org.sirix.api.xml.XmlResourceManager;
+
+import javax.inject.Singleton;
+
+/**
+ * The Dagger component that manages database dependencies. This class is internal and managed by {@link Databases}.
+ *
+ * @author Joao Sousa
+ */
+@Component(modules = DatabaseModule.class)
+@Singleton
+public interface DatabaseManager {
+
+    LocalDatabaseFactory<JsonResourceManager> jsonDatabaseFactory();
```

Comment:
I left some methods without javadocs, as I'm not sure what is the policy here. Feel free to request further docs if you think it is importante.